### PR TITLE
CMake script improvements (compat with s2 v0.11)

### LINKED ISSUE #14: Noisy Tests Build
When build tests is turned on, the following warnings are enabled, which mostly (as far as I can tell) warn about things in the S2, OpenSSL and absl headers.

https://github.com/paleolimbot/s2geography/blob/master/CMakeLists.txt#L176-L178

This makes for a noisy build which probably hides any issues that might be in s2geography. Is there a rationale to keep them?

### PR DESCRIPTION
**Pull request by benbovy**
Some updates to the cmake script(s):

- [x] support both s2geometry versions <0.11 (still using the Finds2.cmake module) and >=0.11 (using the config file installed with s2)
- [x] bump the s2geometry version to 0.11.1 with the `S2_SOURCE=BUNDLE` option.
- [x] silent warnings from s2 headers with the `S2_SOURCE=BUNDLE` option (closes #14)
- [x] add cmake configuration files for reusing s2geography as dependency in downstream cmake projects

@paleolimbot @pramsey I had to add some (quite ugly) workarounds to make #13 work (`S2_VERSION_**` were undefined so the old definition of `S2CellRelation` was still selected even with s2 v0.11). As you mentioned there is still no easy way to get the version of s2 when it is found via `find_package()`. I'm not sure that requiring the user to manually specify the version (e.g., as a cmake option) would be better.

To make things easier to maintain, I'm wondering if we couldn't just drop supporting versions of s2 < 0.11? For my usage (Python / spherely) that would be fine. What about your usage? Any thoughts?

### CODE REVIEW

File: .github/workflows/build-and-run.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
         uses: actions/checkout@v3
```
Comment (martinfleis) [inline .github/workflows/build-and-run.yaml:36]: actions/checkout@v3 shall be bumped to v4

**Comment by benbovy** [comment]
Cool thanks!

Any objections if I cut a 0.1.1 release after merging this PR? It will help me moving forward with packaging in Spherely.

**Comment by benbovy** [comment]
> Any objections if I cut a 0.1.1 release after merging this PR? It will help me moving forward with packaging in Spherely.

I went ahead and published https://github.com/paleolimbot/s2geography/releases/tag/0.1.1 (hope you don't mind!)

**Comment by benbovy** [comment]
And released https://github.com/paleolimbot/s2geography/releases/tag/0.1.2 (sorry!)

**Pull request merged**